src/messages/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,181 @@
+		const shiftTabKeyPress = event.shiftKey && event.key === "Tab";
+
+		// Find last input field of time picker
+		const minutesAsLastTimeInput = !!enableTime && time_24hr;
+		const amPmAsLastTimeInput = !!enableTime && !time_24hr;
+
+		// Time input fields
+		const hourField = webchatWindow?.getElementsByClassName(
+			"flatpickr-hour",
+		)?.[0] as HTMLElement;
+		const minutesField = webchatWindow?.getElementsByClassName(
+			"flatpickr-minute",
+		)?.[0] as HTMLElement;
+		const amPmField = webchatWindow?.getElementsByClassName(
+			"flatpickr-am-pm",
+		)?.[0] as HTMLElement;
+
+		// Check if last time input field is focused
+		const isLastTimeInputFieldFocused =
+			(minutesAsLastTimeInput && event.target === minutesField) ||
+			(amPmAsLastTimeInput && event.target === amPmField);
+
+		// Close Date picker on pressing Escape
+		if (event.key === "Esc" || event.key === "Escape") {
+			handleClose();
+		}
+
+		// Focus should be trapped within date-picker
+		// Handle Tab Navigation
+		if (tabKeyPress) {
+			console.log(event.target);
+			if (event.target === button) {
+				event.preventDefault();
+				calender?.focus(); // Move focus to calender from submit button
+			} else if (isLastTimeInputFieldFocused) {
+				event.preventDefault();
+				button?.focus(); // Move focus to cancel button from last time input field
+			}
+		}
+		// Handle Reverse Tab Navigation
+		if (shiftTabKeyPress) {
+			if (event.target === calender) {
+				event.preventDefault();
+				button?.focus(); // Move focus to Submit button from calender
+			} else if (event.target === hourField) {
+				event.preventDefault();
+				calender?.focus(); // Move focus to calender from hour input field
+			}
+		}
+	};
+
+	return (
+		<>
+			<PrimaryButton onClick={handleOpen} disabled={!!messageParams?.hasReply}>
+				{openText}
+			</PrimaryButton>
+			{showPicker && (
+				<div
+					className={classnames(classes.wrapper, "webchat-plugin-date-picker")}
+					data-testid="datepicker-message"
+					onKeyDown={handleKeyDown}
+					tabIndex={0}
+					role="dialog"
+					aria-modal="true"
+					aria-labelledby={datePickerHeading}
+					aria-describedby={datePickerDescription}
+				>
+					<div
+						className={classnames(classes.header, "webchat-plugin-date-picker-header")}
+					>
+						<span className={mainClasses.srOnly} id={datePickerDescription}>
+							Please use Left/ Right arrows to move focus to previous/ next day.
+							Please use Up/ Down arrows to move focus to the same day of previous/
+							next week. Please use Control + Left/ Right arrows to change the grid of
+							dates to previous/ next month. Please use Control + Up/ Down arrows to
+							change the grid of dates to previous/ next year.
+						</span>
+						<span className={classes.left}></span>
+						<span className={classes.center}>
+							<Typography
+								variant="h2-semibold"
+								component="span"
+								dangerouslySetInnerHTML={{ __html: "Calendar" }}
```

Comment:
why dangerouslySetInnerHTML prop used here?

Reply:
also, it should be `eventName ` from plugin config

Reply:
fixed